modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -921,7 +921,7 @@ public void blockGateways() {
             GridCacheAffinityManager aff = cache.context().affinity();
 
             if (aff != null)
-                aff.cancelFutures();
+                aff.onKernalStop(true);
```

Comment:
Alexey,

Why do you explicitly call aff.onKernalStop() here? This is semantically wrong because onKernalStop is a part of component lifecycle and should not be invoked manually.